### DISCUSSION TITLE
Passing values into slint while using mcu variable

### DISCUSSION THREAD

**Discussion by MohaBeacon** [slint-ui/slint#5288] [Q&A]
any example to increase counter based on push button 
```rust
 let count :i32 = 0;
fn create_slint_app() -> AppWindow {
    let ui = AppWindow::new().expect("Failed to load UI");
    ui.set_counter(ui.get_counter+count);
    ui
}
```
passing a local variable to slint ui out of slint app

**Comment by hunger** [comment]
This should generate `set_counter(int)` and `get_counter() -> int` on `AppWindow`.

```slint
import { LineEdit, HorizontalBox } from "std-widgets.slint";

export component AppWindow {
    in-out property <int> counter: 0;

    preferred-height: 30px;

    layout := HorizontalBox {
        LineEdit {
            enabled: false;
            text: root.counter;
        }
}
```

You can pass local variables back and forth this way as you need.

You could also define a `callback` to call into Rust code as needed. Callbacks are generated on `AppWindow` as `on_<CALLBACK_NAME>(...)`.

Does that help?

**Reply by MohaBeacon** [reply]
no, not this one - i have these version already - what i need is from mcu environment i need to pass the value into slintapp function in rust - then using call back i could push it into ui.

consider this counter is a sensor data. i need to pass it into ui

let count :i32 = 0;
fn create_slint_app() -> AppWindow {
let ui = AppWindow::new().expect("Failed to load UI");
ui.set_counter(ui.get_counter+count);
ui
}

**Reply by hunger** [reply]
Assuming the `AppWindow` code from my last reply, this should work (untested of course):

```rust
fn create_slint_app() -> AppWindow {
    let ui = AppWindow::new().expect("Failed to load UI");
    let count = get_count_from_somewhere();
    ui.set_counter(ui.get_counter() + count);
    ui
}
```

You probably want to keep the actual counter value on the rust side to avoid the `ui.get_counter()` part though.
